src/main/kotlin/com/exactpro/th2/netty/bytebuf/util/ByteBufDataUtil.kt

```diff
@@ -19,69 +19,101 @@
 package com.exactpro.th2.netty.bytebuf.util
 
 import io.netty.buffer.ByteBuf
-import io.netty.buffer.ByteBufUtil
 import java.nio.charset.Charset
 import java.util.*
 
-const val DEFAULT_PAD_CHAR = '\u0000'
+const val DEFAULT_END_CHAR = '\u0000'
 
-fun ByteBuf.readAsciiChar(): Char = this.readCharSequence(1, Charsets.US_ASCII)[0]
+fun ByteBuf.readAsciiChar(): Char = (readByte().toInt() and 0xFF).toChar()
 
-fun ByteBuf.writeAsciiChar(value: Char): ByteBuf = this.apply {
-    require(value.code < 256) {
-        "The $value char with ${value.code} code is out of ASCII table"
-    }
-    ByteBufUtil.writeAscii(this, value.toString())
+fun ByteBuf.writeAsciiChar(value: Char): ByteBuf {
+    require(value.code <= Byte.MAX_VALUE) { "The $value char with ${value.code} code is out of ASCII table" }
+    writeByte(value.code)
+    return this
 }
 
+/**
+ * Reads specified bytes, decodes them to string via specific charset and discards tail after the end character
```

Comment:
We can write additional method with auto timing when it will be required.
I think end character is common approach for binary protocol

Reply:
As far I understand you've added this method to read padded values. If that's the case then we should trim instead of splitting by the first padding character, otherwise it would break in case if string contains padding character somewhere in the middle. Therefore I would've made trimming the default behavior and created a separate method which would split instead of trimming.

Reply:
Split to two methods:
readStringAndDiscard
writePaddedString